driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/fchdir.bpf.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(fchdir_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, FCHDIR_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_FCHDIR_E, FCHDIR_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: fd (type: PT_FD) */
+	s32 fd = (s32)extract__syscall_argument(regs, 0);
+	ringbuf__store_s64(&ringbuf, (s64)fd);
```

Comment:
:+1: Excellent question.. like to learn about it as well.

Reply:
That's a really good question!! This is due to different reasons:
* In kernel <= 4.17 (without raw_tracepoint) syscall arguments were accessible only in the enter event so in very early syscalls probably our choice was influenced by this.
* in some cases sinsp impose some rules on how we can collect our events, for example in events with some flags the param must be in a certain position and with a certain name
* in some cases where syscall are vulnerable to TOCTOU attacks we need to catch parameters both in the enter and in the exit events
* add to it a little bit of confusion due to different contributors that don't know all the sinsp tricky rules (me included) and you have the result :rofl: 

Unfortunately in the modern probe i have to use the same events so we need to keep the same schema of the old probe :(

Reply:
Alright, makes sense, I'll keep an eye out on these things so I can deepen my knowledge on the subject. Thanks Andrea for the explanation!

Reply:
I know this is quite a generic answer, but unfortunately, these tricky rules are not written anywhere... Just to provide you with a quick example: 

https://github.com/falcosecurity/libs/blob/afca72cec1994a5834f9ec2e9687b4be4a1c3c61/userspace/libsinsp/parsers.cpp#L689-L703

If an enter event uses the `EF_USES_FD` flag then the first parameter must be an `fd` otherwise `sinsp` will crash :/

Reply:
Awesome! Thanks a lot for your explanations!